spring-kafka/src/main/java/org/springframework/kafka/annotation/RetryableTopicAnnotationProcessor.java

```diff
@@ -116,12 +122,10 @@ public RetryTopicConfiguration processAnnotation(String[] topics, Method method,
 		boolean traverse = false;
 		if (StringUtils.hasText(annotation.traversingCauses())) {
 			Boolean traverseResolved = resolveExpressionAsBoolean(annotation.traversingCauses(), "traversingCauses");
-			if (traverseResolved != null) {
-				traverse = traverseResolved;
-			}
-			else {
-				traverse = !includes.isEmpty() || !excludes.isEmpty();
-			}
+			traverse = Objects.requireNonNullElseGet(
+					traverseResolved,
+					() -> !includes.isEmpty() || !excludes.isEmpty()
+			);
```

Comment:
the above change is suggested by IDEA. Let me know whether we are happy with the change and I can revert back to old state.